hooks/gen_docs.py

```diff
@@ -1,10 +1,308 @@
+    :returns: A tuple containing lists of all sections and defaults-related sections
+    """
+    component_definition_sections_names = [
+        section.name for section in COMPONENTS_DEFINITION_SECTIONS
+    ]
+    component_file_name = component_name + ".yaml"
+    if exist_changes:
+        component_sections = filter_sections(
+            component_name, component_definition_sections_names, True
+        )
+        component_sections_not_inherited = filter_sections(
+            component_name, component_definition_sections_names
+        )
+        with PATH_DOCS_COMPONENTS_DEPENDENCIES.open("a") as f:
+            yaml.dump({component_file_name: component_sections}, f)
+        with PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS.open("a") as f:
+            yaml.dump({component_file_name: component_sections_not_inherited}, f)
+    else:
+        component_sections: list[str] = PIPELINE_COMPONENT_DEPENDENCIES[  # type: ignore [reportGeneralTypeIssues]
+            component_file_name
+        ]
+        component_sections_not_inherited: list[
+            str
+        ] = DEFAULTS_PIPELINE_COMPONENT_DEPENDENCIES[  # type: ignore [reportGeneralTypeIssues]
+            component_file_name
+        ]
+    return KpopsComponent(component_sections, component_sections_not_inherited)
+
+
+is_change_present = (
+    check_for_changes_in_kpops_component_structure() or is_change_present
+)
+
+try:
+    PIPELINE_COMPONENT_DEPENDENCIES = load_yaml_file(PATH_DOCS_COMPONENTS_DEPENDENCIES)
+    DEFAULTS_PIPELINE_COMPONENT_DEPENDENCIES = load_yaml_file(
+        PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS
+    )
+except OSError:
+    is_change_present = True
+
+for component_file in PIPELINE_COMPONENT_HEADER_FILES:
+    component_file_name = component_file.name
+    component_type = component_file.stem
+    component_defaults_name = "defaults-" + component_file_name
+
+    component_sections, component_sections_not_inheritted = get_sections(
+        component_type,
+        is_change_present,
+    )
+
+    defaults_sections_paths = [
+        PATH_DOCS_RESOURCES / "pipeline-defaults/headers" / component_defaults_name
+    ] + [
+        PATH_DOCS_COMPONENTS / "sections" / section
+        for section in component_sections_not_inheritted
+    ]
+    sections_paths = [component_file] + [
+        PATH_DOCS_COMPONENTS / "sections" / section for section in component_sections
+    ]
+    concatenate_text_files(
+        *(defaults_sections_paths),
+        target=PATH_DOCS_RESOURCES / "pipeline-defaults/" / component_defaults_name,
+    )
+    concatenate_text_files(
+        *(sections_paths),
+        target=PATH_DOCS_RESOURCES / "pipeline-components" / component_file_name,
+    )
+concatenate_text_files(
+    *(
+        component_file.parents[1] / component_file.name
+        for component_file in PIPELINE_COMPONENT_HEADER_FILES
+        if KafkaConnector.get_component_type()
+        != component_file.stem  # Shouldn't be used in the pipeline def
```

Comment:
I would prefer to swap the operands in the condition, because it's the same way in other places aka reads nicer :)
```suggestion
        if component_file.stem
        != KafkaConnector.get_component_type() # shouldn't be used in the pipeline def
```

Reply:
agreed, will do